cache/subtopic.go

```diff
@@ -5,32 +5,66 @@ import (
 	"sync"
 )
 
-// SubtopicsIDs contains a list of subtopics in map form with addition to mutex locking
+// Subtopics contains a list of subtopics in map form with addition to mutex locking
 // The subtopicsMap is used to keep a record of subtopics to be later used to generate the subtopics id `query` for a topic
 // and to check if the subtopic id given by a user exists
-type SubtopicsIDs struct {
+type Subtopics struct {
 	mutex        *sync.RWMutex
-	subtopicsMap map[string]bool
+	subtopicsMap map[string]Subtopic
+}
+
+type Subtopic struct {
+	ID              string
+	LocaliseKeyName string
+	ReleaseDate     string
 }
 
 // GetNewSubTopicsMap creates a new subtopics id map to store subtopic ids with addition to mutex locking
-func NewSubTopicsMap() *SubtopicsIDs {
-	return &SubtopicsIDs{
+func NewSubTopicsMap() *Subtopics {
+	return &Subtopics{
 		mutex:        &sync.RWMutex{},
-		subtopicsMap: make(map[string]bool),
+		subtopicsMap: make(map[string]Subtopic),
 	}
 }
 
-// Get returns a bool value for the given key (id) to inform that the subtopic id exists
-func (t *SubtopicsIDs) Get(key string) bool {
+// Get returns subtopic for given key (id)
+func (t *Subtopics) Get(key string) Subtopic {
 	t.mutex.RLock()
 	defer t.mutex.RUnlock()
 
 	return t.subtopicsMap[key]
 }
 
+// GetSubtopics returns an array of subtopics
+func (t *Subtopics) GetSubtopics(key string) (subtopics []Subtopic) {
+	if t.subtopicsMap == nil {
+		return
+	}
+
+	t.mutex.RLock()
```

Comment:
(minor): I think you are writing to a map here right ? so this needs to be a ```t.mutex.Lock()``` rather than just a read lock.